src/libfetchers/path.cc

```diff
@@ -108,6 +110,18 @@ struct PathInputScheme : InputScheme
         // FIXME: check whether access to 'path' is allowed.
         auto storePath = store->maybeParseStorePath(absPath);
 
+        auto srcSt = lstat(absPath);
```

Comment:
Since you’re using the filesystem api, it might be nicer to use it all the way down and replace `lstat` by something like [last_write_time](https://en.cppreference.com/w/cpp/filesystem/last_write_time)

(depends how clunky it is obviously. I didn’t really try using it)

Reply:
I implemented it in a different fashion now as @edolstra suggested.